lib/Monads/nondet/Nondet_While_Loop_Rules_Completeness.thy

```diff
@@ -260,15 +265,23 @@ proof -
     assume y: "Q r' s'"
     have ?thesis
       using x y
-      apply (induct rule: in_whileLoop_induct)
-       apply (rule_tac x="[(r,s)]" in exI)
-       apply clarsimp
-      apply clarsimp
-      apply (case_tac l)
-       apply clarsimp
-      apply (rule_tac x="(r, s)#l" in exI)
-      apply clarsimp
-      done
+    proof (induct rule: in_whileLoop_induct)
```

Comment:
when followed by `proof` I would usually indent `using` the same as `proof`. Maybe that's not very consistent. What do others think? @Xaphiosis?

Reply:
I have no strong opinions, but the auto-indenter likes it this way. For clarity without the diff formatting getting in the way, this is
```
proof -
  {
    fix r' s'
    assume x: "(r', s') ∈ fst (whileLoop C B r s)"
    assume y: "Q r' s'"
    have ?thesis
      using x y
    proof (induct rule: in_whileLoop_induct)
      case (1 r s)
        apply -
 ```

Reply:
For now, let's not fight the auto indenter on this one. Some of my habits are from before-indenter-times and probably not worth preserving.